apiserver/allfacades.go

```
@@ -3,79 +3,266 @@
+	"github.com/juju/juju/apiserver/machinemanager" // ModelUser Write
+	"github.com/juju/juju/apiserver/machineundertaker"
+	"github.com/juju/juju/apiserver/meterstatus"
+	"github.com/juju/juju/apiserver/metricsadder"
+	"github.com/juju/juju/apiserver/metricsdebug" // ModelUser Write
+	"github.com/juju/juju/apiserver/metricsmanager"
+	"github.com/juju/juju/apiserver/migrationflag"
+	"github.com/juju/juju/apiserver/migrationmaster"
+	"github.com/juju/juju/apiserver/migrationminion"
+	"github.com/juju/juju/apiserver/migrationtarget" // ModelUser Write
+	"github.com/juju/juju/apiserver/modelconfig"     // ModelUser Write
+	"github.com/juju/juju/apiserver/modelmanager"    // ModelUser Write
+	"github.com/juju/juju/apiserver/payloads"
+	"github.com/juju/juju/apiserver/payloadshookcontext"
+	"github.com/juju/juju/apiserver/provisioner"
+	"github.com/juju/juju/apiserver/proxyupdater"
+	"github.com/juju/juju/apiserver/reboot"
+	"github.com/juju/juju/apiserver/remotefirewaller"
+	"github.com/juju/juju/apiserver/remoterelations"
+	"github.com/juju/juju/apiserver/resources"
+	"github.com/juju/juju/apiserver/resourceshookcontext"
+	"github.com/juju/juju/apiserver/resumer"
+	"github.com/juju/juju/apiserver/retrystrategy"
+	"github.com/juju/juju/apiserver/singular"
+	"github.com/juju/juju/apiserver/spaces"    // ModelUser Write
+	"github.com/juju/juju/apiserver/sshclient" // ModelUser Write
+	"github.com/juju/juju/apiserver/statushistory"
+	"github.com/juju/juju/apiserver/storage"
+	"github.com/juju/juju/apiserver/storageprovisioner"
+	"github.com/juju/juju/apiserver/subnets"
+	"github.com/juju/juju/apiserver/undertaker"
+	"github.com/juju/juju/apiserver/unitassigner"
+	"github.com/juju/juju/apiserver/uniter"
+	"github.com/juju/juju/apiserver/upgrader"
+	"github.com/juju/juju/apiserver/usermanager"
+	"github.com/juju/juju/feature"
+	"github.com/juju/juju/state"
 )
+
+// AllFacades returns a registry containing all known API facades.
+func AllFacades() *facade.Registry {
```

Thinking out loud - convention in tests at least would be to name this function MustAllFacades()
(as it panics instead of returning an error)